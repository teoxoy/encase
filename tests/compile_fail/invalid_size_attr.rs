use encase::ShaderType;

fn main() {}

#[derive(ShaderType)]
struct Test {
    #[shader(size)]
    a: u32,
    #[shader(size())]
    b: u32,
    #[shader(size(invalid))]
    c: u32,
    #[shader(size(runtime))]
    d: u32,
    #[shader(size(-1))]
    e: u32,
}
