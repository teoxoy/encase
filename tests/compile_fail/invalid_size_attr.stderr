error: expected attribute arguments in parentheses: `size(...)`
 --> tests/compile_fail/invalid_size_attr.rs:7:14
  |
7 |     #[shader(size)]
  |              ^^^^

error: expected u32 literal or `runtime` identifier
 --> tests/compile_fail/invalid_size_attr.rs:9:19
  |
9 |     #[shader(size())]
  |                   ^

error: expected u32 literal or `runtime` identifier
  --> tests/compile_fail/invalid_size_attr.rs:11:19
   |
11 |     #[shader(size(invalid))]
   |                   ^^^^^^^

error: expected u32 literal or `runtime` identifier
  --> tests/compile_fail/invalid_size_attr.rs:13:19
   |
13 |     #[shader(size(-1))]
   |                   ^
